File: plumbot-bungee/src/main/java/me/regadpole/plumbot/internal/kook/KookClient.java
package me.regadpole.plumbot.internal.kook;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import snw.jkook.command.CommandManager;
import snw.jkook.command.JKookCommand;
import snw.jkook.config.ConfigurationSection;
import snw.kookbc.SharedConstants;
import snw.kookbc.impl.CoreImpl;
import snw.kookbc.impl.KBCClient;
import snw.kookbc.impl.command.internal.PluginsCommand;
import snw.kookbc.impl.entity.builder.EntityBuilder;
import snw.kookbc.impl.entity.builder.MessageBuilder;
import snw.kookbc.impl.event.EventFactory;
import snw.kookbc.impl.network.NetworkClient;
import snw.kookbc.impl.storage.EntityStorage;
import snw.kookbc.interfaces.network.NetworkSystem;
import snw.kookbc.util.ReturnNotNullFunction;

import java.io.File;

public class KookClient extends KBCClient {
    public KookClient(CoreImpl core, ConfigurationSection config, File pluginsFolder, String token) {
        super(core, config, pluginsFolder, token);
    }

    public KookClient(CoreImpl core, ConfigurationSection config, File pluginsFolder, String token, @NotNull String networkMode) {
        super(core, config, pluginsFolder, token, networkMode);
    }

    public KookClient(CoreImpl core, ConfigurationSection config, File pluginsFolder, String token, @Nullable ReturnNotNullFunction<KBCClient, CommandManager> commandManager, @Nullable ReturnNotNullFunction<KBCClient, NetworkClient> networkClient, @Nullable ReturnNotNullFunction<KBCClient, EntityStorage> storage, @Nullable ReturnNotNullFunction<KBCClient, EntityBuilder> entityBuilder, @Nullable ReturnNotNullFunction<KBCClient, MessageBuilder> msgBuilder, @Nullable ReturnNotNullFunction<KBCClient, EventFactory> eventFactory, @Nullable ReturnNotNullFunction<KBCClient, NetworkSystem> networkSystem) {
        super(core, config, pluginsFolder, token, commandManager, networkClient, storage, entityBuilder, msgBuilder, eventFactory, networkSystem);
    }

    @Override
    protected void registerInternal() {
        ConfigurationSection commandConfig = this.getConfig().getConfigurationSection("internal-commands");
        if (commandConfig == null) {
            commandConfig = this.getConfig().createSection("internal-commands");
        }

        if (commandConfig.getBoolean("plugins", true)) {
            this.registerPluginsCommand();
        }
    }

    @Override
    protected void registerStopCommand() {

    }

    @Override
    protected void registerPluginsCommand() {
        (new JKookCommand("plugins")).setDescription("获取已安装到此 " + SharedConstants.IMPL_NAME + " 实例的插件列表。").setExecutor(new PluginsCommand(this)).register(this.getInternalPlugin());
    }

    @Override
    protected void registerHelpCommand() {
    }


}

File: src/main/kotlin/me/regadpole/plumbot/internal/KookClient.kt
package me.regadpole.plumbot.internal

import snw.jkook.command.JKookCommand
import snw.jkook.config.ConfigurationSection
import snw.kookbc.SharedConstants
import snw.kookbc.impl.CoreImpl
import snw.kookbc.impl.KBCClient
import snw.kookbc.impl.command.internal.PluginsCommand
import java.io.File


class KookClient(coreImpl: CoreImpl, config: ConfigurationSection, file: File, token: String, networkMode: String) :
    KBCClient(coreImpl, config, file, token, networkMode) {

    override fun registerInternal() {
        var commandConfig: ConfigurationSection? = config.getConfigurationSection("internal-commands")
        if (commandConfig == null) {
            commandConfig = config.createSection("internal-commands")
        }

        if (commandConfig.getBoolean("plugins", true)) {
            this.registerPluginsCommand()
        }
    }

    override fun registerStopCommand() {
    }

    override fun registerPluginsCommand() {
        (JKookCommand("plugins")).setDescription(("获取已安装到此 " + SharedConstants.IMPL_NAME).toString() + " 实例的插件列表。")
            .setExecutor(
                PluginsCommand(
                    this
                )
            ).register(this.internalPlugin)
    }

    override fun registerHelpCommand() {
    }


}